Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/SimulatedDensityOfStates.py

```diff
@@ -423,60 +427,77 @@ def _compute_partial_ion_workflow(self, partial_ions, frequencies, eigenvectors,
 
         # Output each contribution to it's own workspace
         for ion_name, ions in partial_ions.items():
-            partial_ws_name = self._out_ws_name + '_' + ion_name
+            partial_ws_name = self._out_ws_name + '_'
 
-            self._compute_partial(ions, frequencies, eigenvectors, weights)
+            partial_ws = self._compute_partial(ions, frequencies, eigenvectors, weights)
 
             # Set correct units on partial workspace
-            mtd[self._out_ws_name].setYUnit('(D/A)^2/amu')
-            mtd[self._out_ws_name].setYUnitLabel('Intensity')
+            partial_ws.setYUnit('(D/A)^2/amu')
+            partial_ws.setYUnitLabel('Intensity')
 
             # Add the sample material to the workspace
-            SetSampleMaterial(InputWorkspace=self._out_ws_name,
-                              ChemicalFormula=ion_name)
+            chemical = ion_name
+            if ':' in ion_name:
+                chemical = ion_name.split(':')[0]
+                partial_ws_name += chemical
+                # Parse isotope to rounded int
+                isotope = str(int(round(self._element_isotope[ion_name])))
+                chemical = '(' + chemical + isotope + ')'
+                partial_ws_name += str('('+isotope+')')
```

Comment:
No need for the ```str``` call. ```isotope``` is already a string and if it wasn't I believe this would fail at the ```+``` concat anyway.